### PR TITLE
Use .detach() instead of .remove() for removing dynamic assocs

### PR DESCRIPTION
**Pull request by jeremiahmegel**
From [the jQuery docs for `.detach()`](https://api.jquery.com/detach/#entry-longdesc):

> The `.detach()` method is the same as [`.remove()`](https://api.jquery.com/remove/), except that `.detach()` keeps all jQuery data associated with the removed elements.

Submitting this because we needed to do some extra processing on some data which was set via jQuery's [`.data()`](https://api.jquery.com/data/).

See also: [this Stack Overflow question](https://stackoverflow.com/q/25121168/2384183)

### CODE REVIEW

**Comment by nathanvda** [comment]
Small enough change. Please enlighten me: _why_ did you still need to access `data` from removed items? Just curious 😜

**Comment by jeremiahmegel** [comment]
@nathanvda Thanks! To answer your question, we basically have a `<select>` mirroring some Cocoon associations, so we stored the primary key of the association as a data attribute on each `<option>` and on each Cocoon element. So we listen for `cocoon:after-remove` and de-select the relevant `<option>`. Certainly not the only way to implement that logic, but it fits our use case
